### ISSUE TITLE
Compiler generates invalid code for class B extends (0, A)

### ISSUE THREAD

**Issue by thheller** [google/closure-compiler#4020]
The closure-compiler generates some invalid code when running in `shadow-cljs` and processing the `react-aria` npm package. Or more specifically the `@internationalized/date` package. Specifically [this file](https://unpkg.com/browse/@internationalized/date@3.0.2/dist/main.js) contains a lot of `class X extends (0, Y)` constructs. I do not know why they use `(0, ...)` here and the compiler does indeed complain about "Suspicious code".

The problem can be shortened as such:

Running the latest google-closure-compiler npm via `npx` on this example

```
class A {
}

class B extends (0, A) {
}

console.log(new B());
```

```
npx google-closure-compiler --js bug.js  -O SIMPLE
```
yields
```
bug.js:4:17: WARNING - [JSC_USELESS_CODE] Suspicious code. This code lacks side-effects. Is there a bug?
  4| class B extends (0, A) {
                      ^

0 error(s), 1 warning(s)
class A{}class B extends 0,A{}console.log(new B);
```

Note, the the `()` were removed, which ends up as `class B extends 0,`. Which is invalid code and breaks when trying to load the file. The warning here doesn't help much since I cannot change the npm package.

Is there any way to prevent this from happening?

**Comment by brad4d** [comment]
It seems likely that the logic in the CodePrinter / CodeGenerator isn't handling the priority of the comma operator correctly when it appears as the "extends" child of a class definition.

**Comment by frost-cy** [comment]
Fixed with 80c48e239c8b39e0ed72d68b20b43fcd1fa8125a
